server/src/main/java/org/elasticsearch/inference/InferenceServiceConfiguration.java

```diff
@@ -80,14 +78,11 @@ public InferenceServiceConfiguration(StreamInput in) throws IOException {
     private static final ConstructingObjectParser<InferenceServiceConfiguration, Void> PARSER = new ConstructingObjectParser<>(
         "inference_service_configuration",
         true,
-        args -> {
-            List<String> taskTypes = (ArrayList<String>) args[2];
-            return new InferenceServiceConfiguration.Builder().setService((String) args[0])
-                .setName((String) args[1])
-                .setTaskTypes(EnumSet.copyOf(taskTypes.stream().map(TaskType::fromString).collect(Collectors.toList())))
```

Comment:
`copyOf` throws if it receives an empty set so I modified this to allow an empty set via the builder. An empty set should be unlikely in production because we shouldn't be getting the configuration at all if no task types are supported but it helps the tests.